presto-geospatial-toolkit/src/main/java/com/facebook/presto/geospatial/rtree/IntersectionEnumeration.java

```diff
@@ -0,0 +1,63 @@
+package com.facebook.presto.geospatial.rtree;
+
+import com.esri.core.geometry.ogc.OGCGeometry;
+import com.facebook.presto.geospatial.Rectangle;
+import com.facebook.presto.geospatial.rtree.StrRTree.Node;
+
+import java.util.Enumeration;
+import java.util.Optional;
+import java.util.Spliterator;
+import java.util.Stack;
+import java.util.stream.Stream;
+
+public class IntersectionEnumeration
+        implements Enumeration<OGCGeometry>
+{
+    private final Stack<Node> stack = new Stack<Node>();
```

Comment:
I would use an `ArrayDeque` and use `Deque<Node>`.

Reply:
Any particular reason?  I'm interested in the LIFO behavior, so that I get DFS-like drill down, which keeps the stack size relatively small.